packages/opentelemetry-basic-tracer/test/BasicTracer.test.ts

```diff
@@ -167,6 +167,50 @@ describe('BasicTracer', () => {
       assert.deepStrictEqual(context.traceState, undefined);
     });
 
+    it('Should create real span when not sampled but recording events true', () => {
+      const tracer = new BasicTracer({
+        sampler: NEVER_SAMPLER,
+        scopeManager: new NoopScopeManager(),
+      });
+      const span = tracer.startSpan('my-span', { isRecordingEvents: true });
+      assert.ok(span instanceof Span);
+      assert.strictEqual(span.context().traceOptions, TraceOptions.UNSAMPLED);
+      assert.strictEqual(span.isRecordingEvents(), true);
+    });
+
+    it('Should not create real span when not sampled and recording events  false', () => {
+      const tracer = new BasicTracer({
+        sampler: NEVER_SAMPLER,
+        scopeManager: new NoopScopeManager(),
+      });
+      const span = tracer.startSpan('my-span', { isRecordingEvents: false });
+      assert.ok(span instanceof NoRecordingSpan);
+      assert.strictEqual(span.context().traceOptions, TraceOptions.UNSAMPLED);
+      assert.strictEqual(span.isRecordingEvents(), false);
+    });
+
+    it('Should not create real span when not sampled and no recording events configured', () => {
+      const tracer = new BasicTracer({
+        sampler: NEVER_SAMPLER,
+        scopeManager: new NoopScopeManager(),
+      });
+      const span = tracer.startSpan('my-span');
+      assert.ok(span instanceof NoRecordingSpan);
+      assert.strictEqual(span.context().traceOptions, TraceOptions.UNSAMPLED);
+      assert.strictEqual(span.isRecordingEvents(), false);
+    });
+
+    it('Should create real span when sampled and recording events true', () => {
```

Comment:
is there a real case where we want the span is sampled but not recording events, say due to the tracer being disabled?

Reply:
I don't think so, all sampled span should by default have recorded events true.